design/core_collection_operations.md

```diff
@@ -0,0 +1,175 @@
+# Design of core collection operations
+
+This document describes an inital design of the core, general data collections operations - e.g.
+operations that aren't specific to a particular collection.
+
+## Assumptions
+
+* Collection information can change at any time, and previous versions are not necessarily
+  accessible.
+  * They should be accessible, at least, to database admins, and possibly via admin API methods.
+* Collections can theoretically be reverted to prior versions at any time.
+* We expect < 1000 collections over the lifetime of the collections infrastructure.
+* There may be many versions of collections as they are updated and adjusted over time.
+  * Including adding new data products to collections.
+* Data product data structures / UIs / etc for collections will usually, but not always,
+  be applicable to other collections as well.
```

Comment:
What is an example of the "not always" case here?

Reply:
This is more of a CYA because I don't think I know enough at this point to assert that a particular data structure / UI / API for a data product is 100% going to work for all future collections